### ISSUE TITLE
Use CLOCK_MONOTONIC for mutex waiting on iOS/macOS

### ISSUE THREAD

**Issue by akoeplinger** [dotnet/runtime#73650]
See https://github.com/dotnet/runtime/issues/58737 and related PRs.

`CLOCK_MONOTONIC` is available on ios 10+, but not `pthread_condattr_setclock`. We disable the code on Apple platforms using `BROKEN_CLOCK_SOURCE` in https://github.com/dotnet/runtime/blob/4211dc40d06195682d4b0fc9a220bdb75732d720/src/mono/mono/utils/mono-os-mutex.h#L39-L41

We should look into using `pthread_cond_timedwait` with a `CLOCK_MONOTONIC` argument or alternatively enable `clock_gettime` (supported by newer macOS/iOS releases) on Apple platforms.

**Comment by SamMonoRT** [comment]
Nice to have, but not blocking release. Moving to 9.0.0

**Comment by ivanpovazan** [comment]
Nice to have, but not blocking release. Moving to 10.0.0
